scilpy/tractograms/streamline_operations.py

```diff
@@ -258,34 +258,34 @@ def cut_invalid_streamlines(sft, epsilon=0.001):
     cutting_counter = 0
     for ind in range(len(sft.streamlines)):
         if ind in indices_to_cut:
-            # This streamline was detected as invalid
-
-            best_pos = [0, 0]  # First and last valid points of longest segment
-            cur_pos = [0, 0]   # First and last valid points of current segment
-            for pos, point in enumerate(sft.streamlines[ind]):
-                if (point < epsilon).any() or \
-                        (point >= sft.dimensions - epsilon).any():
-                    # The coordinate is < 0 or > box. Starting new segment
-                    cur_pos = [pos+1, pos+1]
-                elif cur_pos[1] - cur_pos[0] > best_pos[1] - best_pos[0]:
-                    # We found a longer good segment.
-                    best_pos = cur_pos
-
-                # Ready to check next point.
-                cur_pos[1] += 1
-
-            # Appending the longest segment to the list of streamlines
-            if not best_pos == [0, 0]:
-                new_streamlines.append(
-                    sft.streamlines[ind][best_pos[0]:best_pos[1]])
-                cutting_counter += 1
+
+            in_vol = np.logical_and(
+                sft.streamlines[ind] >= epsilon,
+                sft.streamlines[ind] < sft.dimensions - epsilon).all(axis=1)
+
+            # Get segments in the streamline that are within the volume using
+            # ndi.label
+            blobs, _ = ndi.label(in_vol)
```

Comment:
@frheault @arnaudbore I have greatly simplified and sped up the invalid-cutting process. Let me know how you feel about it.